File: scala/src/main/scala/cn/spark/scala/Top3.scala
package scala.cn.spark.scala

import org.apache.spark.{SparkConf, SparkContext}

/**
  * Created by wangzhilong on 2017/6/5.
  */
object Top3 {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
      .setAppName("Top3")
      .setMaster("local")
    val sc = new SparkContext(conf)
    val lines = sc.textFile("d://spark//top.txt")
    lines.distinct()
      .map(line => (line.toInt,line))
      .sortByKey(false)
      .map(line => line._2)
      .take(3)
      .foreach(println(_))
  }

}

File: java/src/main/java/cn/spark/core/Top3.java
package cn.spark.core;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import scala.Tuple2;

/**
 * Created by wangzhilong on 2017/6/5.
 */
public class Top3 {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf()
                .setAppName("Top3")
                .setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);

        JavaRDD<String> lines = sc.textFile("d://spark//top.txt");
        lines.distinct()
                .mapToPair(line -> new Tuple2<Integer, String>(Integer.valueOf(line),line))
                .sortByKey(false)
                .take(3)
                .forEach(l -> System.out.println(l._2));
        sc.close();
    }
}
